core/src/impl/Kokkos_Atomic_Load.hpp

```diff
@@ -125,8 +125,7 @@ __device__ __inline__ T _relaxed_atomic_load_impl(
                                     void const**>::type = nullptr) {
   T rv{};
   // TODO remove a copy operation here?
-  Kokkos::Impl::atomic_oper_fetch(NoOpOper<T>{}, &rv, rv);
-  return rv;
+  return Kokkos::Impl::atomic_oper_fetch(NoOpOper<T>{}, ptr, rv);
```

Comment:
There was no point on declaring `rv` in the first place, you could pass `*ptr` instead.

Reply:
not sure if that does an unneccessary read?

Reply:
I would expect the compiler would inline the apply and see it is not used.
Anyway I am not asking you to change.